android/app/src/main/java/app/organicmaps/editor/OsmLoginFragment.java

```diff
@@ -75,12 +95,63 @@ private void login()
     });
   }
 
+  private void loginWithBrowser()
+  {
+    enableInput(false);
+
+    String[] oauthParams = OsmOAuth.nativeOAuthParams();
+    Uri oauth2url = Uri.parse(oauthParams[0] + "/oauth2/authorize")
```

Comment:
If om://oauth2 scheme is parsed in C++, does it make sense to make the initial URI also on the C++ side? Will it allow reusing of a similar login approach for iOS?

Reply:
You are right. I modified to build OAuth2 URL using native code instead of Java.